src/network/networkd-manager-bus.c

```diff
@@ -47,6 +48,75 @@ static int method_list_links(sd_bus_message *message, void *userdata, sd_bus_err
         return sd_bus_send(NULL, reply, NULL);
 }
 
+static int method_get_link_by_name(sd_bus_message *message, void *userdata, sd_bus_error *error) {
+        _cleanup_(sd_bus_message_unrefp) sd_bus_message *reply = NULL;
+        _cleanup_free_ char *path = NULL;
+        Manager *manager = userdata;
+        bool found = false;
+        const char *name;
+        Iterator i;
+        Link *link;
+        int r;
+
+        r = sd_bus_message_read(message, "s", &name);
+        if (r < 0)
+                return r;
+
+        HASHMAP_FOREACH(link, manager->links, i)
+                if (streq_ptr(link->ifname, name)) {
+                        found = true;
+                        break;
+                }
```

Comment:
hmm, i'd just translate the name via if_nametoindex() and then look it up by index. Sounds faster to me if we have a myriad of devices